### PR TITLE
Added local packages search for static feeds

### LINKED ISSUE #607: No working Search for static feeds
Hi! Thanks for reporting this feature/bug/question! 

Please keep / fill in the relevant info from this template so that we can help you as best as possible.

**Type** (choose one):

- Bug

**NPE version**: (e.g. 5.x) (see Help -> about)

5.0.132+f33e9c233d

**OS version**: (Windows 7, 10 - 1803/17134, etc)

Windows 10

**Installed from**: Microsoft Store

***Repro**
Open feed https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json.

Enter `Arcade` into search box and hit enter.

After a while the package list refreshes but contains all packages, not just those that include "Arcade".

### PR DESCRIPTION
**Pull request by 304NotModified**
Including prelease filter, id filter, paging and ranked sorting. Fixes #607

Tested with https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json

Initial load is still a bit slow (~16 sec), but after that search and paging is fast!

(yes, this won't work with static feeds if they have <= 15 packages)

### CODE REVIEW

File: PackageViewModel/PackageSearch/LocalPackageSearcher.cs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                    .OrderByDescending(it => it.score)
+                    .Select(it => it.package);
+            }
+
+            return packages;
+        }
+
+        private static int CalcScore(T package, string searchText)
+        {
+            var score = ScoreForEquals(package.Identity.Id, searchText);
```
Comment (304NotModified) [inline PackageViewModel/PackageSearch/LocalPackageSearcher.cs]: FYI This isn't fully following https://github.com/NuGetPackageExplorer/NuGetPackageExplorer/issues/607#issuecomment-517950958

File: PackageViewModel/PackageSearch/PackageListCache.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using NuGet.Protocol.Core.Types;
+
+namespace PackageExplorerViewModel.PackageSearch
+{
+    internal class PackageListCache<T>  where T : IPackageSearchMetadata
```
Comment (304NotModified) [inline PackageViewModel/PackageSearch/PackageListCache.cs]: Better cache proposals welcome :)

File: PackageViewModel/PackageSearch/SearchContext.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using NuGet.Protocol.Core.Types;
+
+namespace PackageExplorerViewModel.PackageSearch
+{
+    public class SearchContext
```
Comment (304NotModified) [inline PackageViewModel/PackageSearch/SearchContext.cs]: FYI: Wrapper for current search context for easy passing deeper

File: PackageViewModel/PackageChooser/PackageChooserViewModel.cs [hunk 4]
```diff
@@ -370,15 +372,6 @@ private void ShowPackages(IEnumerable<IPackageSearchMetadata> packages, int pack
             if (ActiveRepository != null)
             {
                 var ar = ActiveRepository;
-                var skip = beginPackage - 1;
```
Comment (304NotModified) [inline PackageViewModel/PackageChooser/PackageChooserViewModel.cs]: FYI, not needed anymore. This is done in another (and better) place

File: PackageViewModel/PackageSearch/LocalPackageSearcher.cs (hunk 1 above)
Comment (onovotny) [reply PackageViewModel/PackageSearch/LocalPackageSearcher.cs]: What are the differences?

Comment (onovotny) [reply PackageViewModel/PackageSearch/LocalPackageSearcher.cs]: Can the package scoring be parellelized to speed it up? Parallel.ForEach, etc?

Comment (304NotModified) [reply PackageViewModel/PackageSearch/LocalPackageSearcher.cs]: Maybe, but it's now already blazing fast. (instant experience with 1900 packages). The slow part is already done (download JSON  + parse all JSON).

**Comment by 304NotModified** [comment]
When we are there, please squashed PR merge :)

**Pull request merged**